### PR TITLE
 Docker enhancements

### PR DESCRIPTION
**Pull request by antonkap**
### Describe the Pull Request

```
Added possibility to use docker secrets inside container.
Added docker-compose_swarm.yml for deployment in docker swarm mode on single node.
Fixed postgres volume mount in docker-compose.yml
```

### Checklist for PR

- [ ] Run MobSF unit tests and lint `tox -e lint,test`
- [ ] Tested Working on Linux, Mac, Windows, and Docker
- [ ] Add unit test for any new Web API (Refer: `StaticAnalyzer/tests.py`)
- [ ] Make sure tests are passing on your PR [MobSF tests](https://github.com/MobSF/Mobile-Security-Framework-MobSF/actions)

### Additional Comments (if any)

```
DESCRIBE HERE
```

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ services:
     image: "postgres:latest"
     restart: always
     volumes:
-      - $HOME/MobSF/postgresql_data:/var/lib/postgresql
+      - $HOME/MobSF/postgresql_data:/var/lib/postgresql/data
```
Comment (ajinabraham) [inline docker-compose.yml:6]: Why explicitly mount `data`?

File: docker-compose_swarm.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      - mobsfDB_password
+  mobsf:
+    environment:
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD_FILE=/run/secrets/mobsfDB_password
+      - POSTGRES_DB=mobsf
+      - POSTGRES_HOST=postgres
+      - POSTGRES_PORT=5432
+      - MOBSF_API_KEY_FILE=/run/secrets/mobsf_api_key
+    image: ${MOBSF_IMAGE:-opensecurity/mobile-security-framework-mobsf:v4.0.7}
```
Comment (ajinabraham) [inline docker-compose_swarm.yml:24]: can this be pointing to latest?

File: docker-compose_swarm.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+version: '3.8'
+services:
+  postgres:
+    image: "postgres:${POSTGRES_VERSION:-17.0-bookworm}"
```
Comment (ajinabraham) [inline docker-compose_swarm.yml:4]: can this be pointing to latest?

Comment (antonkap) [reply docker-compose_swarm.yml:4]: I afraid that newer version can contain some breaking changes. Also it seems like a well-known anti-pattern.
There are some articles about it:
https://www.docker.com/blog/docker-best-practices-using-tags-and-labels-to-manage-docker-image-sprawl/
https://medium.com/@mehar.chand.cloud/docker-10-anti-patterns-what-to-avoid-980fa13d8951#c3f1
https://stackoverflow.com/a/72891342

File: docker-compose_swarm.yml (hunk 2 above)
Comment (antonkap) [reply docker-compose_swarm.yml:24]: I thought that it's better to refer to specific version in specific release. In this way it should be more consistent. Also it's similar to postgres(I've left corresponding comments below)

File: docker-compose.yml (hunk 1 above)
Comment (antonkap) [reply docker-compose.yml:6]: I've just tested it and found that without such modification it doesn't work. As I remember mounted folder was empty after db creation.
From official documentation(https://hub.docker.com/_/postgres):
image

Similar issue: https://stackoverflow.com/a/41650891

Comment (ajinabraham) [reply docker-compose.yml:6]: Sounds good.

File: docker-compose_swarm.yml (hunk 2 above)
Comment (ajinabraham) [reply docker-compose_swarm.yml:24]: We prefer to use the latest image for all examples. It makes sense for posrgres, but for mobsf image, please tag the latest.

**Comment by ajinabraham** [comment]
These will be included in https://github.com/MobSF/Mobile-Security-Framework-MobSF/pull/2445/commits/8e57a788b49224a8749e9457c7da2620e10bd48e

**Comment by ajinabraham** [comment]
Changes merged in https://github.com/MobSF/Mobile-Security-Framework-MobSF/pull/2445

**Pull request closed without merging**